File: leetcode/go/912.sort-an-array.go
func sortArray(nums []int) []int {
    counts := make([]int, 100001)
    for _, n := range nums {
        counts[50000+n]++
    }
    ans := make([]int, 0, len(nums))
    for i, count := range counts {
        for _ = range count {
            ans = append(ans, i-50000)
        }
    }
    return ans
}

File: leetcode/cpp/912.sort-an-array.cpp
class Solution {
public:
    vector<int> sortArray(vector<int>& nums) {
        vector<int> counts(100001, 0);
        for (int n: nums) {
            counts[50000+n]++;
        }
        vector<int> ans(nums.size());
        int k = 0;
        for (int i = 0; i < 100001 && k < nums.size(); i++) {
            for(int j = 0; j < counts[i]; j++) {
                ans[k] = i - 50000;
                k++;
            }
        }
        return ans;
    }
};

